drafts/angular-navigation-concept.md

```diff
@@ -0,0 +1,436 @@
+
+Since we have the `currentPageId` we just compare it to the already updated `ɵrouterPageId`.
+If it `ɵrouterPageId` is smaller than `currentPageId`, we know it is a back navigation.
+![How to detect the navigation direction](https://cdn.hashnode.com/res/hashnode/image/upload/v1722076281194/vqj69u35t.png?auto=format)
+
+However, important to note is, this only works with `popstate`.
+Imperative navigations (navigations that were made with `router.navigate`) will always be forward navigations.
+
+Alright, now we know how to detect back navigations.
+We can now update the `guard` from before.
+```ts
+export const guard = (childRoute) => {
+  ...
+  if (navigationSerivce.isBackNavigation() && (data['backSkip'] as boolean) ?? false) {
+    navigationService.navigateAfterSkipped();
+    return false;
+  }
+  return true;
+}
+
+@Injectable({ providedIn: 'root' })
+export class NavigationService {
+  private navigationDirectionService = inject(NavigationDirectionService);
+  ...
+  isBackNavigation() {
+    return this.navigationDirectionService.getNavigationType() === 'back';
+  }
+}
+```
+
+And just like this, our second requirement is fulfilled.
+Super simple right?
+
+## Navigate back to hub pages
+This is the simplest of all requirements.
+We already have the logic, to navigate to flow source pages.
+Let us use this code and extend it to work with hub pages.
+
+```ts
+import { Location } from '@angular/common';
+
+type NavigationTarget = 'flow-source-page' | 'hub-page';
+
+@Injectable({ providedIn: 'root' })
+export class NavigationService {
+  ...
+  
+  isNavigationTarget(routeData: Data) {
+    if(this.navigationTarget === 'flow-source-page') {
+      return (routeData['flowSourcePage'] as boolean) ?? false;
+    }
+    if(this.navigatedToTarget() === 'hub-page') {
+      return (routeData['hubPage'] as boolean) ?? false;
+    }
+    throw new Error('isNavigationTarget should never be called when navigationTarget is undefined');
+  }
+  
+  ...
+}
+```
+
+Then we just have to mark the hub pages in the route data.
+```ts
+{
+    path: '/hubpage',
+    component: HubPageComponent,
+    pathMatch: 'full',
+    data: { hubPage: true }
+}
+```
+
+And to trigger the navigation to the hub page, we just have to call `navigationService.navigateTo('hub-page')`.
+And that is it, we did it and it works almost flawless.
+
+
+# Known Issues
+Remember how `NavigationSkipped` works?
+Well yeah me too, and we have one big problem because of it.
+Let us have a look at this example.
+![Navigation that breaks our navigation concept](https://cdn.hashnode.com/res/hashnode/image/upload/v1722077740906/XsuS1iHDB.png?auto=format)
+We want to navigate from the last `/a` to the `hub page`.
+But we get stuck at the first `/a`.
+The reason for this is, it is the same URL, and our config says we should ignore those cases.
+Meaning, the guards are not called anymore and only `NavigationSkipped` is dispatched.
+So far I have not found a solution to this problem.
```

Comment:
maybe you can add a shoutout to comment this article if anyone has an idea how to resolve it.